### DISCUSSION TITLE
Error when using server actions: backingFormData.forEach is not a function

### DISCUSSION THREAD

**Discussion by frasermarch** [vercel/next.js#49552] [App Router]
Whenever I try and use server action I get the following error:

Error: backingFormData.forEach is not a function

parseModelString
.../node_modules/next/dist/compiled/react-server-dom-webpack-experimental/cjs/react-server-dom-webpack-server.node.development.js (2865:27)

I'm in a completely new project using npx create-next-app@latest, enabling server actions in next config, and changing page.js to:

```
export default function Form() {
    async function handleSubmit() {
      "use server"
      console.log("submit")
    }
   
   
    return (
      <form action={handleSubmit}>
        <button type="submit">Submit</button>
      </form>
    );
  }
```

Any help would be really appreciated!

**Comment by frasermarch** [comment]
I needed to update node. I was on v18. Not sure if this is somewhere in the docs but I feel it should be

**Comment by technorav3nn** [comment]
This NEEDS to be mentioned on the docs. I spent an hour trying to fix it.

**Reply by frasermarch** [reply]
Turns out it is, on the installation page https://nextjs.org/docs/getting-started/installation

**Reply by mortona42** [reply]
> System Requirements:
  [Node.js 16.8](https://nodejs.org/) or later.

Why doesn't 18 work?
